### PR TITLE
feat(algo): add cispo_anchor to select old vs rollout IS-ratio anchor

### PR DESCRIPTION
**Pull request by tbalestri-lila**
## What

Add `cispo.cispo_anchor: "old" | "rollout"` to select which behavior policy the CISPO clamped importance-sampling ratio is anchored on. Default `"old"` preserves current behavior exactly.

## Why

CISPO currently anchors its clamped IS ratio on the recomputed old log-probs (π_θ / π_old), where π_old is recomputed fresh by the training backend at experience-prep. That ratio therefore only deviates from 1 once π_θ has moved away from π_old — i.e. only when the batch takes **more than one gradient update** (`update_epochs > 1`, or `train_batch_size > policy_mini_batch_size` so there are multiple minibatch steps).

Fully-async training takes exactly **one** gradient step per batch (it hard-asserts `train_batch_size == policy_mini_batch_size` and `update_epochs == 1`), so at loss time π_θ == π_old exactly → ratio ≡ 1 → the CISPO clamp never bites. Stock CISPO is effectively inert under fully-async.

With `cispo_anchor="rollout"` the ratio is anchored on the rollout/sampler log-probs (π_θ / π_rollout) instead. Unlike π_old, π_rollout is **not** a fresh backend recompute, so the ratio deviates from 1 even at a single gradient step, from two async-intrinsic sources: (1) staleness — the rollout was sampled by an older policy version than the current trainer weights; and (2) sampler/trainer engine mismatch — vLLM logprobs differ from the training-backend logprobs even for identical weights. So the clamped objective genuinely engages under async. This is the clamped (cap-not-zero) counterpart of the existing `rollout_is` loss, which anchors on π_rollout but HARD-ZEROS out-of-range tokens; CISPO caps them so every token keeps a bounded gradient.

## Notes

- Default `"old"` is a no-op — existing runs are unchanged.
- The rollout-anchored ratio is itself the off-policy correction, so stacking TIS (`off_policy_correction.tis_ratio_type`) is rejected to avoid double-counting.
- `"rollout"` requires `rollout_logprobs` (asserted); the trainer's skip-fwd path is made anchor-aware so the unused old-logprobs forward is skipped when the rollout anchor makes it unnecessary.

## Tests

- `test_losses.py`: rollout-anchor loss value, the `rollout_logprobs` requirement, the TIS double-count guard, and config validation.
- `test_skip_fwd_logprobs.py`: the cispo+rollout skip-path.

### CODE REVIEW

File: skyrl/backends/skyrl_train/utils/ppo_utils.py [hunk 1]
```diff
@@ -744,16 +744,65 @@ def compute_policy_loss_cispo(
...
+    # IS-ratio diagnostics. clip_ratio above is only the cap-HIT fraction; these expose the actual
+    # gradient multiplier r = pi_theta / pi_behavior. Under cispo_anchor="rollout", r drifts from 1
+    # via the vLLM-vs-train numerical gap + async staleness even at a single gradient pass, so mean(r)
+    # and the raw-vs-clamped gap quantify how much the off-policy correction (and the cap) actually
+    # move the gradient -- not just how often the cap fires. The `* loss_mask` zeros masked tokens for
+    # max/min; masked_mean excludes them for the means.
+    cispo_ratio_mean = masked_mean(ratio, loss_mask).mean().detach().item()
+    cispo_ratio_clamped_mean = masked_mean(clamped_ratio, loss_mask).mean().detach().item()
+    cispo_ratio_max = (ratio * loss_mask).max().detach().item() if loss_mask is not None else ratio.max().item()
+    cispo_ratio_min = (ratio * loss_mask).min().detach().item() if loss_mask is not None else ratio.min().item()
```
[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl/backends/skyrl_train/utils/ppo_utils.py]: This is important. Can you address this @tbalestri-lila ?

**Review by SumanthRH — COMMENTED** [review]
LGTM pending the ratio min computation fix

File: skyrl/backends/skyrl_train/utils/ppo_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbalestri-lila) [reply skyrl/backends/skyrl_train/utils/ppo_utils.py]: Fixed in 8a911d52. `ratio` is strictly positive (`safe_exp_delta`), so `(ratio * loss_mask).min()` was reporting `0.0` whenever any token was masked. Now indexing the boolean mask to take min/max over active ratios only, and added a regression test (`test_policy_loss_cispo_ratio_min_max_ignore_masked_tokens`) that masks a token and asserts min/max reflect just the unmasked ratios — verified it fails against the old line.

**Review by SumanthRH — APPROVED** [review]
